src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -170,9 +184,13 @@ final case class PurposeApiServiceImpl(
           Future.failed(MissingFreeOfChargeReason)
         else Future.unit
       purpose        <- purposeManagementService.getPurposeById(purposeUUID)
+      _              <- catalogManagementService.getEServiceById(purposeUpdateContent.eserviceId)
```

Comment:
as a minor optimization, we can skip this call.
As per row 192, if the eserviceId is the same, we are sure that it exists, if the id has changed, we check if an agreement exists (and if there is an agreement, the eservice must exist)